tests/sampling/test_forward.py

```diff
@@ -1801,3 +1806,114 @@ def test_sample_prior_predictive_samples_deprecated_warns() -> None:
     match = "The samples argument has been deprecated"
     with pytest.warns(DeprecationWarning, match=match):
         pm.sample_prior_predictive(model=m, samples=10)
+
+
+@pytest.fixture(params=["deterministic", "observed", "conditioned_on_observed"])
+def variable_to_vectorize(request):
+    if request.param == "deterministic":
+        return ["y"]
+    elif request.param == "conditioned_on_observed":
+        return ["z", "z_downstream"]
+    else:
+        return ["z"]
+
+
+@pytest.fixture(params=["allow_rvs_in_graph", "disallow_rvs_in_graph"])
+def allow_rvs_in_graph(request):
+    if request.param == "allow_rvs_in_graph":
+        return True
+    else:
+        return False
+
+
+@pytest.fixture(scope="module", params=["nested_random_variables", "no_nested_random_variables"])
+def has_nested_random_variables(request):
+    return request.param == "nested_random_variables"
+
+
+@pytest.fixture(scope="module")
+def model_to_vectorize(has_nested_random_variables):
+    with pm.Model() as model:
+        if not has_nested_random_variables:
+            x_parent = 0.0
+        else:
+            x_parent = pm.Normal("x_parent")
+        x = pm.Normal("x", mu=x_parent)
+        d = pm.Data("d", np.array([1, 2, 3]))
+        obs = pm.Data("obs", np.ones_like(d.get_value()))
+        y = pm.Deterministic("y", x * d)
+        z = pm.Gamma("z", mu=pt.exp(y), sigma=pt.exp(y) * 0.1, observed=obs)
+        pm.Deterministic("z_downstream", z * 2)
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore")
+        with model:
+            idata = pm.sample(100, tune=100, chains=1, cores=1)
+    return freeze_dims_and_data(model), idata
+
+
+@pytest.fixture(params=["rv_from_posterior", "resample_rv"])
+def input_rv_names(request, has_nested_random_variables):
+    if request.param == "rv_from_posterior":
+        if has_nested_random_variables:
+            return ["x_parent", "x"]
+        else:
+            return ["x"]
+    else:
+        return []
+
+
+def test_vectorize_over_posterior(
```

Comment:
Nitpick, I really don't like fixtures used like this, makes reading the test behavior very complex. Why not just have if elses in the test function based on the parametrization. Also define the model in the test function?

Reply:
Agree to disagree here. Having the fixture makes it very easy to add a new orthogonal condition that then gets automatically combined with the existing ones. I understand that the test might be harder to follow, but it makes it easier to write extend the test coverage to new scenarios while ensuring that everything else still works fine.

Reply:
I don't see how you can extend and still not have the source of truth (whatever you want to assert) on the test itself, which means you still need to know what all the parametrization and their combinations do. You just make it harder to read because information is spread all over the place.

To be clear I'm a big fan of parametrize, just no via fixtures. But it's personal preference.

Reply:
I understand. I personally like it with fixtures.